lib/puppet_x/puppetlabs/powershell/powershell_manager.rb

```diff
@@ -144,7 +144,15 @@ def self.init_path
 
       def make_ps_code(powershell_code, timeout_ms = nil, working_dir = nil, environment_variables = [])
         begin
+
+          # Zero timeout is a special case. Other modules sometimes treat this
+          # as an infinite timeout. We don't support infinite, so for the case
+          # of a user specifying zero, we sub in the default value of 300
+          # seconds.
+          if (timeout_ms == 0) then timeout_ms = 300 * 1000 end
```

Comment:
This seems really far down the callstack given we already perform a calculation for seconds to milliseconds in the provider code at 
https://github.com/puppetlabs/puppetlabs-powershell/blob/a7a7b25faefeb513c998243004ab12f30e484fcd/lib/puppet/provider/exec/powershell.rb#L91

That would give us the opportunity to emit a warning to the user via `Puppet.warning` as well.

It actually seems best served to correct this value in the `munge` for the type, but since the type is defined elsewhere, we're not afforded that opportunity. Seems like provider is the next best spot.

Had you considered that approach instead?

Reply:
I considered higher in the stack, but decided to put it as close to the eventual preparation of the PowerShell code as possible. It’s a habit from my database days I guess where the general rule is that a constraint on the way data should be handled should happen as close to the storage and retrieval of the data as possible.